detox/src/client/actions/SyncStatusFormatter.js

```diff
@@ -0,0 +1,277 @@
+    const resourceName = jsonDescription.name;
+    const properties = jsonDescription.description;
+
+    switch (resourceName) {
+      case 'delayed_perform_selector':
+        return SyncStatusFormatter.delayedPerformSelectorResourceDescriptionFromProperties(properties);
+
+      case 'dispatch_queue':
+        return SyncStatusFormatter.dispatchQueueResourceDescriptionFromProperties(properties);
+
+      case 'run_loop':
+        return SyncStatusFormatter.runLoopResourceDescriptionFromProperties(properties);
+
+      case 'one_time_events':
+        return SyncStatusFormatter.oneTimeEventDescriptionFromProperties(properties);
+
+      case 'timers':
+        return SyncStatusFormatter.timersResourceDescriptionFromProperties(properties);
+
+      case 'ui':
+        return SyncStatusFormatter.uiResourceDescriptionFromProperties(properties);
+
+      case 'js_timers':
+        return SyncStatusFormatter.jsTimersResourceDescriptionFromProperties(properties);
+
+      case 'network':
+        return SyncStatusFormatter.networkResourceDescriptionFromProperties(properties);
+
+      case 'looper':
+        return SyncStatusFormatter.looperResourceDescriptionFromProperties(properties);
+
+      case 'io':
+        return SyncStatusFormatter.resourceTitle(`Disk I/O activity.`);
+
+      case 'unknown':
+        return SyncStatusFormatter.unknownResourceDescriptionFromProperties(properties);
+
+      default:
+        throw new DetoxRuntimeError(`Given sync status is invalid, cannot find resource name: \`${resourceName}\``);
+    }
+  }
+
+  static delayedPerformSelectorResourceDescriptionFromProperties(properties) {
+    const pendingSelectors = SyncStatusFormatter.getPropertyFromObject('pending_selectors', properties);
+    return SyncStatusFormatter.resourceTitle(`There are ${pendingSelectors} pending delayed selectors to be ` +
+      `performed.`);
+  }
+
+  static getPropertyFromObject(propertyName, object) {
+    const property = object[propertyName];
+    if (property === undefined) {
+      throw new DetoxRuntimeError(`Given sync status is invalid. Cannot find \`${propertyName}\` property for ` +
+        `resource description`);
+    }
+
+    return property;
+  }
+
+  static resourceTitle(string) {
+    return `• ${string}`;
+  }
+
+  static dispatchQueueResourceDescriptionFromProperties(properties) {
+    const workItemsCount = SyncStatusFormatter.getPropertyFromObject('works_count', properties);
+    const queue = SyncStatusFormatter.getPropertyFromObject('queue', properties);
+    return SyncStatusFormatter.resourceTitle(`There are ${workItemsCount} work items pending on the ` +
+      `dispatch queue: "${queue}".`);
+  }
+
+  static runLoopResourceDescriptionFromProperties(properties) {
+    const runLoopName = SyncStatusFormatter.getPropertyFromObject('name', properties);
+    return SyncStatusFormatter.resourceTitle(`Run loop "${runLoopName}" is awake.`);
+  }
+
+  static oneTimeEventDescriptionFromProperties(properties) {
```

Comment:
The entire object looks static.

Please just use `const` and `function` instead of having a class.

Reply:
great, done